src/control_ids_resolver.py

```diff
@@ -0,0 +1,66 @@
+import boto3, logging, http.client, json, botocore
+
+
+class ControlIdsResolver:
+
+    def __init__(self, security_controls, parameter_name_api_host, parameter_name_api_key,
+                 parameter_name_api_resource_path):
+        self.local_security_control_id = security_controls.replace(' ', '').split(',')
+        self._parameter_name_api_host = parameter_name_api_host
+        self._parameter_name_api_key = parameter_name_api_key
+        self._parameter_name_api_resource_path = parameter_name_api_resource_path
+        self._ssm = boto3.client('ssm')
+        self._ssm_param_root_path = "/SecurityReporter"
+        self._get_param_store_configs()
+        self._ignore_api = self._is_api_config_valid()
+        self._centralized_security_control_ids = self._get_centralized_security_controls()
+
+    def _get_param_store_configs(self) -> dict:
+        try:
+            response = self._ssm.get_parameters_by_path(
+                Path=self._ssm_param_root_path,
+                WithDecryption=True
+            )
+            parameters = response['Parameters']
+            self._control_ids_api_details = {parameter['Name']: parameter['Value'] for parameter in parameters if
+                                             parameter['Name'] in [self._parameter_name_api_host, self._parameter_name_api_key,
+                                                           self._parameter_name_api_resource_path]}
+        except botocore.exceptions.ClientError as e:
+            logging.exception(e.response)
+            self._control_ids_api_details = dict()
+
+    def _is_api_config_valid(self):
+        if 'NOT_SET' in self._control_ids_api_details.values():
+            logging.warning(f"setup for control-ids api is considered not valid. {self._control_ids_api_details}")
+            return False
+        return True
+
+    def _get_centralized_security_controls(self):
+        logging.warning(f"api details for obtaining control ids {self._control_ids_api_details}")
+        if not self._ignore_api:
+            logging.warning(f"skipping obtaining data from control-id api.")
+            return []
+        try:
+            headers = {
+                'x-api-key': self._control_ids_api_details[self._parameter_name_api_key],
+                'Content_Type': 'application/json'
+            }
+            connection = http.client.HTTPSConnection(self._control_ids_api_details[self._parameter_name_api_host])
```

Comment:
There is connection object made but there is no check if parameter_name_api_host is empty. I would check first if this has a meaningful value and decide if we should make an effort to call API. IF its empty we can automatically go pre defined constants in lambda array.